code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -563,16 +563,19 @@
 
 /obj/effect/timestop/Initialize()
 	. = ..()
-	for(var/M in GLOB.living_mob_list)
+	for(var/M in GLOB.player_list)
 		var/mob/living/L = M
-		for(var/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop/T in L.mind.spell_list) //People who can stop time are immune to timestop
-			immune |= L
+		for(var/V in L.mind.spell_list) //People who can stop time are immune to its effects
+			var/obj/O = V
+			if(istype(O, /obj/effect/proc_holder/spell/aoe_turf/conjure/timestop))
```

Comment:
just do `if(locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in L.mind.spell_list)`